pkg/rancher-desktop/backend/k3sHelper.ts

```diff
@@ -678,7 +685,21 @@ export default class K3sHelper extends events.EventEmitter {
 
           sums[filename] = sum;
         }
-        const promises = [this.filenames.exe, this.filenames.images].map(async(filename) => {
+
+        let existsIndex;
+
+        for (let index = 0; typeof existsIndex === 'undefined' && index < this.filenames.images.length; index++) {
+          try {
+            await fs.promises.access(path.join(dir, this.filenames.images[index]), fs.constants.R_OK);
+            existsIndex = index;
```

Comment:
If we `break` after setting `existsIndex`, then we don't need that `typeof` check in the loop.

Which makes it something like:
```ts
for (const [index, imageName] of Object.entries(this.filenames.images)) {
  try {
    await fs.promises.access(path.join(dir, imageName), fs.constants.R_OK);
    existsIndex = index;
    break;
  } catch {
    // Ignore access errors and try next iteration, if any.
  }
}
```
(Approving anyway, because at this point this is just bike shedding and the code you have is fine.)